platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/EmbeddedSolrFactory.java

```diff
@@ -31,33 +32,45 @@
 import org.apache.solr.core.SolrCore;
 import org.apache.solr.core.SolrResourceLoader;
 import org.apache.solr.schema.IndexSchema;
+import org.codice.solr.factory.SolrClientFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.xml.sax.InputSource;
 import org.xml.sax.SAXException;
 
-public class EmbeddedSolrFactory extends SolrClientFactory {
+import com.google.common.util.concurrent.Futures;
 
-    public static final String DEFAULT_EMBEDDED_CORE_NAME = "embedded";
+public class EmbeddedSolrFactory implements SolrClientFactory {
 
-    public static final String IMMEMORY_SOLRCONFIG_XML = "solrconfig-inmemory.xml";
+    protected static final Logger LOGGER = LoggerFactory.getLogger(EmbeddedSolrFactory.class);
 
-    public static final String DEFAULT_SOLR_XML = "solr.xml";
+    public static final String DEFAULT_EMBEDDED_CORE_NAME = "embedded";
 
-    /**
-     * Hiding constructor
-     */
-    private EmbeddedSolrFactory() {
+    public static final String IMMEMORY_SOLRCONFIG_XML = "solrconfig-inmemory.xml";
 
+    @Override
+    public Future<SolrClient> newClient(String core) {
+        ConfigurationStore configStore = ConfigurationStore.getInstance();
+        if (System.getProperty("solr.data.dir") != null) {
```

Comment:
The default is `${ddf.home}/data/solr`.  That default is overridden if `solr.data.dir` is set.